### ISSUE TITLE
Release Rainbow Padding code/artifacts on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [quasar529/rainbow-padding#1]
Hi @quasar529 🤗

Niels here from the open-source team at Hugging Face. I discovered your work on Arxiv and was wondering whether you would like to submit it to [hf.co/papers](https://hf.co/papers) to improve its discoverability. If you are one of the authors, you can submit it at https://huggingface.co/papers/submit.

The paper page lets people discuss about your paper and lets them find artifacts about it (your models, datasets or demo for instance), you can also claim
the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

I noticed in your abstract that the code for "Rainbow Padding" is stated to be publicly available at `https://github.com/quasar529/rainbow-padding`. However, it appears the repository currently doesn't contain a `README.md` file, suggesting the code might not be fully uploaded or organized yet.

It would be great to make the code for Rainbow Padding, and any fine-tuned model checkpoints (e.g., existing dLLMs improved with Rainbow Padding as mentioned in your paper), available on the 🤗 Hub. This would greatly improve their discoverability and visibility within the AI community. We can add tags so that people find them when filtering https://huggingface.co/models.

## Uploading models (if applicable)

See here for a guide: https://huggingface.co/docs/hub/models-uploading.

In this case, we could leverage the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to any custom `nn.Module`. Alternatively, one can leverage the [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file) one-liner to download a checkpoint from the hub.

We encourage researchers to push each model checkpoint to a separate model repository, so that things like download stats also work. We can then also link the checkpoints to the paper page.

Let me know if you're interested or need any help regarding this, or if you're still working on getting the repository ready!

Cheers,

Niels
ML Engineer @ HF 🤗

**Comment by quasar529** [comment]
Hi @NielsRogge , thanks a lot for your message and for taking the time to look into my work 🤗

I’d like to give you a quick update:
- The implementation has been fully uploaded to the repository
- I’ve also pushed the trained checkpoints to the Hugging Face Hub, so models are already available for download and use: [rainbow-padding-llada](https://huggingface.co/quasar529/rainbow-padding-llada)

I truly appreciate your suggestions and support. Improving the accessibility and visibility of research artifacts within the community is of great value, and I am glad to contribute to this process.  

Best regards, 
BumJun

**Comment by NielsRogge** [comment]
Great! https://huggingface.co/papers/2510.03680 has a linked model which is great.

Feel free to submit your paper to [Daily Papers](https://huggingface.co/papers/submit), this will make more people discover your work.
